components/datastores/add-knowledge-modal.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useState } from 'react';
+import { InboxOutlined } from '@ant-design/icons';
+import type { UploadProps } from 'antd';
+import { message, Upload } from 'antd';
+import { useTranslation } from 'react-i18next';
+import CheckCircleOutlinedIcon from '@mui/icons-material/CheckCircleOutlined';
+import { Modal, Button, Sheet, Stack, Box, Input, Textarea, Switch, Typography, styled } from '@/lib/mui';
+import { sendSpacePostRequest, sendSpaceUploadPostRequest } from '@/utils/request';
+
+const { Dragger } = Upload;
+
+const Item = styled(Sheet)(({ theme }) => ({
+  width: '33%',
+  backgroundColor: theme.palette.mode === 'dark' ? theme.palette.background.level1 : '#fff',
+  ...theme.typography,
+  padding: theme.spacing(1),
+  textAlign: 'center',
+  borderRadius: 4,
+  color: theme.vars.palette.text.secondary,
+}));
+
+interface IProps {
+  isAddKnowledgeSpaceModalShow: boolean;
+  setIsAddKnowledgeSpaceModalShow: (isAddKnowledgeSpaceModalShow: boolean) => void;
+  setKnowledgeSpaceList: (list: Array<any>) => void;
+}
+
+export default function AddKnowledgeModal(props: IProps) {
+  const { t } = useTranslation();
+
+  const stepsOfAddingSpace = [t('Knowledge_Space_Config'), t('Choose_a_Datasource_type'), t('Setup_the_Datasource')];
+
+  const [owner, setOwner] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [webPageUrl, setWebPageUrl] = useState<string>('');
+  const [knowledgeSpaceName, setKnowledgeSpaceName] = useState<string>('');
+  const [documentType, setDocumentType] = useState<string>('');
+  const [activeStep, setActiveStep] = useState<number>(0);
+  const [synchChecked, setSynchChecked] = useState<boolean>(true);
+  const [documentName, setDocumentName] = useState<any>('');
+  const [textSource, setTextSource] = useState<string>('');
+  const [originFileObj, setOriginFileObj] = useState<any>(null);
+  const [text, setText] = useState<string>('');
+
+  const uploadProps: UploadProps = {
+    name: 'file',
+    multiple: false,
+    beforeUpload: () => false,
+    onChange(info) {
+      if (info.fileList.length === 0) {
+        setOriginFileObj(null);
+        setDocumentName('');
+        return;
+      }
+      setOriginFileObj(info.file);
+      setDocumentName(info.file.name);
+    },
+  };
+  const documentTypeList = [
```

Comment:
`documentTypeList` is declared in serval files, please declare it in `constants.ts`